### DISCUSSION TITLE
about pyodide loading

### DISCUSSION THREAD

**Discussion by ChinaGodzilla** [pyscript/pyscript#1910] [General]
Using "pyscript", I found that the test page loaded very slowly. I found that "Pyodide" was downloaded when the page was loading. This speed is unacceptably slow. How can I deploy "Pyodide" locally and let the page load local files?

**Comment by WebReflection** [comment]
Pyodide bootstraps in 1 to X seconds, accordingly with the hardware, no matter if local or CDN, it's the Pyodide bootstrap itself that is complex and surely slow, compared to MicroPython.

Add micropip packages in the mix, and bootstrap increase X times over regular bootstrap before anything gets executed, even if you download all available packages locally.

One approach is to use MicroPython on the main or move Pyodide in a worker so that at least the rest of the page remains interactive.

There are various examples around how to show a spinner or a loading info so if that's your cup of tea, I can provide some link/example to do that.

Unfortunately we don't get to decide the bootstrap speed of Pyodide, we can only suggest alternative approaches or solution to that slow bootstrap.

**Comment by ChinaGodzilla** [comment]
@WebReflection  thank you
